src/main/scala/com/atomist/rug/runtime/js/interop/NashornMapBackedGraphNode.scala

```diff
@@ -65,12 +66,23 @@ private class NashornMapBackedGraphNode(som: ScriptObjectMirror,
   val nodeId: Option[String] = relevantPropertiesAndValues.get(NodeIdField).map(id => "" + id)
   nodeRegistry.register(this)
 
+  private val traversableEdges: Map[String,Seq[GraphNode]] =
+    relevantPropertiesAndValues.keys
+      .map(key => (key, toNode(key)))
+      .toMap
+
   override def nodeName: String = relevantPropertiesAndValues.get(NodeNameField) match {
     case None => ""
     case Some(s: String) => s
     case x => Objects.toString(x)
   }
 
+  override def address: String = relevantPropertiesAndValues.get(NodeAddressField) match {
+    case None => null
```

Comment:
An empty string isn't an out of band value. It means the root of the path. I agree that using a possible `null` return is ugly. Think I may be able to fix this via a mixin in `NashornGraphNode`.